add 0.8.36 release announcement blogpost (#207)
ethereum/solidity-website: src/posts/2026-07-09-solidity-0.8.36-release-announcement.md

---
title: "Solidity 0.8.36 Release Announcement"
date: "2026-07-09"
author: Solidity Team
category: Releases
---

[Solidity v0.8.36](https://github.com/argotorg/solidity/releases/tag/v0.8.36) is now available.

> The release includes two security fixes, both of medium severity, each with its own detailed write-up below.

Reviewer: The intro is what we use for the blurbs in various places. I'd keep it concise.
```suggestion
The release includes two security fixes, both of medium severity.
```

On the experimental side, the SSA-form code generator introduced in 0.8.35 gains stack-to-memory spilling, which effectively solves stack-too-deep on that backend.

> The experimental EOF backend is also removed; it had been unused since EOF was dropped from its target network upgrade.

Reviewer: The intro is what we use for the blurbs in various places. I'd keep it concise.

As for EOF, it was never really used so this did not change. It's just that no one needs it now.

```suggestion
The release fixes two important bugs of medium severity.
On the experimental side, the SSA-form code generator introduced in 0.8.35 gains stack-to-memory spilling, which effectively solves stack-too-deep on that backend.
The experimental EOF backend is also removed; it had been obsolete since EOF was rejected for inclusion in the Fusaka network upgrade.
```

## Notable Features and Changes

> ### Security Fixes

Reviewer: ```suggestion
## Important Bugfixes
```

Then '## Notable Features and Changes' should go below this section.

Author: One could argue that a security bugfix is a notable feature and change :)

Reviewer: This is just how we've always been doing it. I still have that idea to change how we handle miscompilations and drop the "important bugfix" terminology. But until then I'd keep using it not to muddle things up even further.

- **Stack-to-memory mover can reuse live memory.**
  A flaw in how the Yul optimizer detects recursive call cycles could misclassify a recursive function as non-recursive, letting the stack-too-deep mitigation move variables into memory slots that are later overwritten.
  Write-up: https://blog.soliditylang.org/2026/07/09/unsound-spill-in-mutual-recursion-bug/
- **Custom storage layout can reverse inheritance order.**
  A warning check for storage layouts placed near the end of storage could accidentally reverse a contract's inheritance linearization, causing base contracts and their constructors to run in the wrong order (present since 0.8.29).

>   Write-up: https://blog.soliditylang.org/2026/07/09/inheritance-order-reversal-on-storage-end-warning-bug/

Reviewer: Links should have titles.

I'd also tweak these a bit to include slightly more detail, so that people have an easier time deciding if the bug is relevant to them. Especially the bold parts sound very broad, making these bugs seem more severe than they are.

```suggestion
- **Stack-to-memory mover can mistakenly apply spilling to recursive functions.**
    A flaw in how the Yul optimizer detects call cycles could misclassify mutually recursive functions as non-recursive.
    This allowed the stack-too-deep mitigation move their variables to memory, despite the fact that multiple invocations of the same function would use the same memory slots and overwrite each other's values.
    Write-up: [Unsound Spill In Mutual Recursion Bug](https://blog.soliditylang.org/2026/07/09/unsound-spill-in-mutual-recursion-bug/)
- **Custom storage layout can corrupt inheritance order annotation when very close to the end of storage.**
    A bug in the code issuing a warning about contract's variables being placed too close to the end of storage could accidentally reverse the AST annotation describing the linearization of contract's inheritance hierarchy.
    This would affect analysis and code generation in a multitude of ways, most notably, changing the initialization order of state variables and the invocation order of constructors and virtual functions.
    Write-up: [Inheritance Order Reversal On Storage End Warning Bug](https://blog.soliditylang.org/2026/07/09/inheritance-order-reversal-on-storage-end-warning-bug/)
```

Reviewer: I also still think that we should generally start just using the bug list entries directly in the release blog post. They're already meant to concisely summarize the bugs.

### SSA CFG Stack-to-Memory Spilling (Experimental)

Solidity 0.8.35 [introduced an experimental SSA-form control-flow-graph code generator](https://soliditylang.org/blog/2026/04/29/solidity-0.8.35-release-announcement/#experimental-ssa-cfg-code-generator) for the IR pipeline.
One of the gaps we noted at the time was that stack-to-memory spilling had not yet been ported to the SSA backend.

This release adds it.

> With spilling in place, the code generator can move values to memory when the stack runs out of room, so code that the existing pipelines reject with a "stack too deep" error now compiles through the SSA backend.

Reviewer: Let's settle on one way to refer to this bug :) I don't mind which one as long as it's the same one - the post uses `stack-too-deep` in the intro.

```suggestion
With spilling in place, the code generator can move values to memory when the stack runs out of room, so code that the existing pipelines reject with a stack-too-deep error now compiles through the SSA backend.
```

It remains experimental, enabled with `--experimental --via-ssa-cfg`, and is not yet the default for `--via-ir`.
Stabilizing the SSA CFG code generator is one of our main priorities for the next six months.

### Removal of the Experimental EOF Backend

0.8.36 removes the experimental EOF (EVM Object Format) backend.
A substantial amount of EOF support had been implemented in anticipation of its inclusion in an upcoming network upgrade, but EOF was dropped from that upgrade late in the process and the code has been sitting unused since.

> Removing it is routine housekeeping.

Reviewer: ```suggestion
A substantial amount of EOF support had been implemented in anticipation of its inclusion in the Fusaka network upgrade, but the feature was ultimately rejected from that upgrade late in the process.
With no mainnet support and no real prospect of adoption in by any network in the future the backend is obsolete.
As a matter of routine housekeeping we have decided to finally remove this code to simplify the code base.
```
